### ISSUE TITLE
ryujinx: SDL motion input doesn't work

### ISSUE THREAD

**Issue by SuperSamus** [NixOS/nixpkgs#230927]
### Describe the bug
Ryujinx, due to using the SDL2 backend in order to communicate with the game controller, supports motion input (like the gyroscope) for the controllers that support it (like the Nintendo Switch Pro Controller).
However, when using Ryujinx built from nixpkgs, motion input doesn't work, as if the controller you are using doesn't support it (even if it does).
If instead you use Ryujinx downloaded from [the official website](https://ryujinx.org/download/), then motion input will work.

### Steps To Reproduce
1. Open Ryujinx;
2. Configure a controller that supports motion input, make sure that motion input is enabled;
3. Open a game that uses motion input;
4. Move the controller when the game requires it;

### Expected behavior
Moving the controller should have an effect in the game, which is what happens when downloading Ryujinx from the official website.

### Additional context
Tested on both GTK3 and Avalonia GUI.
I'm not talking about Cemuhook.

### Notify maintainers

@IvarWithoutBones @06kellyjac

### Metadata
```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.2.12-xanmod1, NixOS, 23.05 (Stoat), 23.05.20230503.0d373d5`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by 06kellyjac** [comment]
Do you have any logs that would be useful here? I was under the impression the SDL2 dotnet dependency just handles this as the GH actions build of upstream Ryujinx doesn't bring anything special.

**Comment by SuperSamus** [comment]
NOTE: `steam-run` can't launch Ryujinx anymore due to #231533.

So, here are the logs:
[logs-ryujinx-nixpkgs.txt](https://github.com/NixOS/nixpkgs/files/11478228/logs-ryujinx-nixpkgs.txt)
[logs-ryujinx-source.txt](https://github.com/NixOS/nixpkgs/files/11478230/logs-ryujinx-source.txt)

Unfortunately, I can't notice any significant difference.

What I did notice, though:
If I launch Ryujinx from one place, set the gamepad I want to use, then launch the other Ryujinx, it forgets the gamepad I want to use and I have to set it again (the mapping is remembered, only "which controller" is forgotten). Which is weird, since they have the same ID.
That's why in both logs you can see `GUI.WindowThread Hid Remap: No matching controllers found`, because after I launched the game, I went to Ryujinx's settings to set the gamepad.

**Comment by SuperSamus** [comment]
> So the ryujinx that *was* working with motion controls was also on nixos ran via `steam-run`?

Yes.

Also, another thing: Dolphin also supports motion input with SDL[^1], and that one also works even when compiled from NixOS.

[^1]:  If compiled with `ENABLE_SDL`.

**Comment by lasers** [comment]
Can you confirm that Dolphin still works today with motion input?  I asked because I remember seeing it worked on NixOS, but I can't get it to work right now. I see `evdev/0/Nintendo Switch Pro Controller` in Device dropdown, but no IMU.

**Comment by SuperSamus** [comment]
You need to use SDL, not evdev.
Anyway, still works today.

To test it yourself, build Dolphin with:
```
nix-build -E 'with import <nixpkgs> {}; pkgs.dolphin-emu.overrideAttrs(old: {buildInputs = old.buildInputs ++ [ SDL2 ]; cmakeFlags = old.cmakeFlags ++ [ "-DENABLE_SDL=ON" ];})'
```

**Comment by SuperSamus** [comment]
1. `hid-nintendo` is included in the kernel since 5.16, you should not install it manually.
2. I'm not using joycond (among the things it does, is adding an udev rule to expose the IMU sensors.), I instead lazily put myself in the `input` group.
3. I don't want to sound offensive, but you are making noise (the evdev stuff is irrelevant to the issue). Ask help in their official Discord server https://discord.dolphin-emu.org/.

**Comment by SuperSamus** [comment]
This is fixed now (I presume by #299646).
